pkg/utils/k8sutil.go

```diff
@@ -471,184 +267,94 @@ func GetReadyPod(pods *v1.PodList) (v1.Pod, error) {
 	return v1.Pod{}, errors.New("There is no pod ready")
 }
 
-// extract the branch number from the runtime string
-func getBranchFromRuntime(runtime string) string {
-	re := regexp.MustCompile("[0-9.]+")
-	return re.FindString(runtime)
+func appendToCommand(orig string, command ...string) string {
+	if len(orig) > 0 {
+		return fmt.Sprintf("%s && %s", orig, strings.Join(command, " && "))
+	}
+	return strings.Join(command, " && ")
 }
 
-// specify image for the init container
-func getInitImage(runtime string) string {
-	switch {
-	case strings.Contains(runtime, "python"):
-		branch := getBranchFromRuntime(runtime)
-		if branch == "2.7" {
-			// TODO: Migrate the image for python 2.7 to an official source (not alpine-based)
-			return "tuna/python-pillow:2.7.11-alpine"
-		}
-		return "python:" + branch
-	case strings.Contains(runtime, "nodejs"):
-		return "node:6.10-alpine"
-	case strings.Contains(runtime, "ruby"):
-		return "bitnami/ruby:2.4"
-	case strings.Contains(runtime, "dotnetcore"):
-		return "microsoft/aspnetcore-build:2.0"
+func getProvisionContainer(function, checksum, fileName, handler, contentType, runtime string, runtimeVolume, depsVolume v1.VolumeMount) (v1.Container, error) {
+	prepareCommand := ""
+	originFile := path.Join(depsVolume.MountPath, fileName)
+	depName, _ := kubelessRuntime.GetRuntimeDepName(runtime)
+
+	// Prepare Function file and dependencies
+	switch contentType {
+	case "base64":
+		// File is encoded in base64
+		prepareCommand = appendToCommand(prepareCommand, fmt.Sprintf("cat %s | base64 -d > %s", originFile, originFile))
```

Comment:
unnecessary use of `cat` - just redirect `base64` stdin directly.
Also: Reading and writing to the same file in the same pipeline will probably do the wrong thing since the shell will truncate the (existing) output file.